api/sys_raft.go

```diff
@@ -43,13 +46,46 @@ func (c *Sys) RaftJoin(opts *RaftJoinRequest) (*RaftJoinResponse, error) {
 // writes it to the supplied io.Writer.
 func (c *Sys) RaftSnapshot(snapWriter io.Writer) error {
 	r := c.c.NewRequest("GET", "/v1/sys/storage/raft/snapshot")
+	r.URL.RawQuery = r.Params.Encode()
 
-	ctx, cancelFunc := context.WithCancel(context.Background())
-	defer cancelFunc()
+	req, err := http.NewRequest(http.MethodGet, r.URL.RequestURI(), nil)
+	if err != nil {
+		return err
+	}
+
+	req.URL.User = r.URL.User
+	req.URL.Scheme = r.URL.Scheme
+	req.URL.Host = r.URL.Host
+	req.Host = r.URL.Host
+
+	if r.Headers != nil {
+		for header, vals := range r.Headers {
+			for _, val := range vals {
+				req.Header.Add(header, val)
+			}
+		}
+	}
+
+	if len(r.ClientToken) != 0 {
+		req.Header.Set(consts.AuthHeaderName, r.ClientToken)
+	}
+
+	if len(r.WrapTTL) != 0 {
+		req.Header.Set("X-Vault-Wrap-TTL", r.WrapTTL)
+	}
+
+	if len(r.MFAHeaderVals) != 0 {
+		for _, mfaHeaderVal := range r.MFAHeaderVals {
+			req.Header.Add("X-Vault-MFA", mfaHeaderVal)
+		}
+	}
+
+	if r.PolicyOverride {
+		req.Header.Set("X-Vault-Policy-Override", "true")
+	}
 
-	// Avoid error checking done by reading out the response, by calling an
-	// internal function
-	resp, err := c.c.rawRequestWithContextInternal(ctx, r)
+	client := &http.Client{}
```

Comment:
You'll likely need to add TLS information here too

Reply:
Very good observation. Thanks for pointing this out. Fixed. Let me know if the changes look alright.